crates/evm/evm/src/inspectors/stack.rs

```diff
@@ -292,6 +302,20 @@ pub struct InspectorStack {
     pub inner_context_data: Option<InnerContextData>,
 }
 
+struct InspectorStackRefMut<'a> {
+    pub cheatcodes: Option<&'a mut Cheatcodes>,
+    pub inner: &'a mut InspectorStackInner,
+}
+
+impl CheatcodesExecutor for InspectorStackInner {
+    fn get_inspector<'a, DB: DatabaseExt>(
+        &'a mut self,
+        cheats: &'a mut Cheatcodes,
+    ) -> impl InspectorExt<DB> + 'a {
+        InspectorStackRefMut { cheatcodes: Some(cheats), inner: self }
+    }
+}
```

Comment:
separating the cheatcodes from the stack makes it possible to pass the `Inner` to the cheatcodes so it can access it during inspection